elements/lisk-p2p/src/p2p.ts

```diff
@@ -497,7 +517,14 @@ export class P2P extends EventEmitter {
 
 		discoveredPeers.forEach((peerInfo: P2PDiscoveredPeerInfo) => {
 			const peerId = constructPeerIdFromPeerInfo(peerInfo);
-			if (!this._triedPeers.has(peerId) && !this._newPeers.has(peerId)) {
+			if (
+				!this._triedPeers.has(peerId) &&
+				!this._newPeers.has(peerId) &&
+				!(
+					NODE_HOST_IP_LIST.includes(peerInfo.ipAddress) &&
+					peerInfo.wsPort === this.nodeInfo.wsPort
```

Comment:
This condition wouldn't work in a production environment. The node's own IP will not be `127.0.0.1` (nor `localhost`), it will look like a regular IP address. The only way to know which IP is our own is during the handshake when we get the nonce.

To make it work properly, we would need to add separate logic for self-discovery and the node would store its own IP address somewhere.

Given the amount of complexity involved, I think it would be simpler to just allow the node to add itself as a peer.

Reply:
It's just an extra check to avoid adding these IPs in `newPeers` list which are showing up sometimes, if it adds itself with its public IP then it's fine

Reply:
same as above